src/pages/password-recovery/password-recovery.component.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react';
+import { toast } from 'react-toastify';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import { makeStyles } from '@material-ui/core/styles';
+import { useHistory } from 'react-router-dom';
+import * as yup from 'yup';
+
+import { passwordRecovery } from 'service/password';
+
+const useStyles = makeStyles((theme) => ({
+  getBackIcon: {
+    marginTop: theme.spacing(4),
+    cursor: 'pointer',
+  },
+  paper: {
+    marginTop: theme.spacing(8),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(1),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+}));
+
+const RecoverPassword = () => {
+  const classes = useStyles();
+  const [email, setEmail] = useState('');
+  const [formError, setError] = useState(false);
+  const [recoverPasswordError, setRecoverPasswordError] = useState('');
+  const [loading, setLoading] = useState(false);
+  const history = useHistory();
+
+  const onStart = () => setLoading(true);
+  const onEnd = () => setLoading(false);
+  const onSuccess = (data) => {
+    toast.success('E-mail de recuperação enviado com sucesso!');
+  };
+
+  const schema = yup.object().shape({
+    email: yup.string().email('e-mail inválido').required(),
+  });
+
+  const onError = (message) => {
+    toast.error(
+      'Não foi possível enviar o e-mail de recuperação, por favor tente novamente mais tarde!'
+    );
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    if (!email) {
+      setError(true);
+      return;
+    }
+
+    const payload = { email };
+
+    await passwordRecovery(payload, onSuccess, onError, onEnd, onStart);
```

Comment:
Keep the request functions in the service layer and consume them using `react-query`. This way you don't need to handle states manually.